src/MultiRegion/multi_region_boundary_conditions.jl

```diff
@@ -59,17 +60,18 @@ fill_halo_regions!(c::MultiRegionObject, ::Nothing, args...; kwargs...) = nothin
 # fill_halo_regions!(c::MultiRegionObject, bcs, loc, mrg::MultiRegionGrid, buffers, args...; kwargs...) = 
 #     apply_regionally!(fill_halo_regions!, c, bcs, loc, mrg, Reference(c.regional_objects), Reference(buffers.regional_objects), args...; kwargs...)
 
-# This results in too large parameter space required on the GPU for too many regions!! (we are passing the whole buffers and regions)
 function fill_halo_regions!(c::MultiRegionObject, bcs, indices, loc, mrg::MultiRegionGrid, buffers, args...; kwargs...) 
 
-    arch = architecture(mrg)
+    arch  = architecture(mrg)
+    halo_tuple  = construct_regionally(permute_boundary_conditions, bcs)
 
-    halo_tuple = construct_regionally(permute_boundary_conditions, bcs)
+    neighbors = Reference(c.regional_objects)
+    buff      = Reference(buffers.regional_objects)
 
     for task = 1:3
         barrier = device_event(arch)
         apply_regionally!(fill_halo_event!, task, halo_tuple, 
-                          c, indices, loc, arch, barrier, mrg, Reference(c.regional_objects), Reference(buffers.regional_objects), 
+                          c, indices, loc, arch, barrier, mrg, neighbors, buff, 
```

Comment:
Irrelevant to this PR but note that `Reference` is easily confused with `Base.Ref`